runtime/src/common.rs

```diff
@@ -41,3 +46,33 @@ pub const MAXIMUM_BLOCK_WEIGHT: Weight = Weight::from_parts(
 	WEIGHT_REF_TIME_PER_SECOND.saturating_mul(2),
 	polkadot_primitives::MAX_POV_SIZE as u64,
 );
+
+/// Logic for the author to get a portion of fees.
+pub struct ToAuthor;
+impl OnUnbalanced<NegativeImbalance<Runtime>> for ToAuthor {
+	fn on_nonzero_unbalanced(amount: NegativeImbalance<Runtime>) {
+		if let Some(author) = <pallet_author_reward_dest::Pallet<Runtime>>::author() {
+			Balances::resolve_creating(&author, amount);
+		} else {
+			log::warn!("Author reward destination not available, unable to reward author.");
+		}
```

Comment:
i think because we are working with reserve dot token, we cannot let any funds not be handled.

So this should put funds into treasury, or deal with fees should detect if there is no author and put 100 to treasury in that logic.

Otherwise, there is some token that we control, but we no longer have representation in our system of

Reply:
I guess though runtime panics when author is not available, but better to handle 100% here too. I could imagine a future where we do not require the author inherent to be provided.

Reply:
yes done: log error and fallback to treasury: https://github.com/polkadot-dropit/node/pull/11/commits/9fc847d48a18cff810fa8f7229fb73c5cbba9e5d